src/components/CartItem.tsx

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable max-len */
+import { getFullPrice } from '../helpers/getFullPrice';
+import { Product } from '../types/Product';
+
+interface Props<T> {
+  product: Product,
+  save: (value: T) => void,
+  products: Product[],
+  setTotalPrice: (value: number) => void,
+  totalPrice: number,
+}
+
+enum Action { add, substract }
+
+export const CartItem: React.FC<Props<Product[]>> = (
+  {
+    product, save, products, setTotalPrice, totalPrice,
+  },
+) => {
+  const handleNumberChange = (action: Action) => {
+    if (action === Action.substract) {
+      if (product.number === 1) {
+        return;
```

Comment:
What do you think about removing the product in this case? You can move the callback from `cart_close` button to a separate function and use it here and in the mentioned button. (optional).
Or disable the button (add the corresponding styles in this case)